packages/core/crates/core-path/src/channel_graph.rs

```diff
@@ -90,73 +59,340 @@ impl ChannelGraph {
         self.me
     }
 
+    /// Looks up an `Open` or `PendingToClose` channel given the source and destination.
+    /// Returns `None` if no such edge exists in the graph.
+    pub fn get_channel(&self, source: Address, destination: Address) -> Option<&ChannelEntry> {
+        self.graph.edge_weight(source, destination).map(|w| &w.channel)
+    }
+
+    /// Gets all `Open` outgoing channels going from the given `Address`
+    pub fn open_channels_from(&self, source: Address) -> impl Iterator<Item = (Address, Address, &ChannelEdge)> {
+        self.graph
+            .edges_directed(source, Direction::Outgoing)
+            .filter(|c| c.weight().channel.status == ChannelStatus::Open)
+    }
+
+    /// Checks whether there's any path via Open channels that connects `source` and `destination`
+    /// This does not need to be necessarily a multi-hop path.
+    pub fn has_path(&self, source: Address, destination: Address) -> bool {
+        let only_open_graph = EdgeFiltered::from_fn(&self.graph, |e| e.weight().channel.status == ChannelStatus::Open);
+        has_path_connecting(&only_open_graph, source, destination, None)
+    }
+
     /// Inserts or updates the given channel in the channel graph.
     /// Returns a set of changes if the channel was already present in the graphs or
     /// None if the channel was not previously present in the channel graph.
     pub fn update_channel(&mut self, channel: ChannelEntry) -> Option<Vec<ChannelChange>> {
-        let weighted = WeightedChannel {
-            channel,
-            weight: 1_f64, // TODO: compute weight properly
-        };
-
-        if let Some(old_w_value) = self.graph.add_edge(channel.source, channel.destination, weighted) {
-            let mut ret = Vec::new();
-            let old_channel = old_w_value.channel;
-
-            if old_channel.status != channel.status {
-                ret.push(Status {
-                    old: old_channel.status,
-                    new: channel.status,
-                });
-            }
-
-            if old_channel.balance != channel.balance {
-                ret.push(CurrentBalance {
-                    old: old_channel.balance,
-                    new: channel.balance,
-                });
-            }
+        // Remove the edge since we don't allow Closed channels
+        if channel.status == ChannelStatus::Closed {
```

Comment:
This is where the prometheus metrics should go then?

Reply:
Seems like a good place yes